skytemple_randomizer/randomizer/npc.py

```diff
@@ -128,42 +124,60 @@ def _smart_replace_text(self, mapped_actor_names_by_lang):
             )
             csk_replace_regions = [
                 self.static_data.string_index_data.string_blocks.get(
-                    "Job Debriefing Related Strings"
+                    "Job Debriefing Related Strings (Secondary)"
                 ),
                 self.static_data.string_index_data.string_blocks.get(
-                    "(MAROWAK-DOJO-STRS-UNMAPPED)"
+                    "Game Trade Strings"
                 ),
                 self.static_data.string_index_data.string_blocks.get(
                     "Spinda's Juice Bar Strings"
                 ),
                 self.static_data.string_index_data.string_blocks.get(
-                    "(CHIMECHO-ASM-STR-UNMAPPED)"
+                    "Chimecho Assembly Strings"
                 ),
                 self.static_data.string_index_data.string_blocks.get(
-                    "Game and Dungeon Hints"
+                    "Mime Jr. Spa Strings"
                 ),
                 self.static_data.string_index_data.string_blocks.get(
-                    "Mime Jr. Spa Strings"
+                    "Adventure Log Entries"
                 ),
                 self.static_data.string_index_data.string_blocks.get(
-                    "Adventure Log Strings"
+                    "Floor-Wide Status"
                 ),
-            ]
-            # Kecleon needs extra care, because some item long descriptions contain references to the shop (should be replaced) and the Pokemon itself (should not be replaced) at the same time.
-            # Instead we replace some of the mentions of the specific text "Kecleon's Shop" (note: things like music track names should not be replaced).
-            kecleon_shop_text = {
-                # IMPORTANT: match.group(1) should always be Kecleon's name
-                "English": re.compile(r"(Kecleon)(?:\[CR])?'s\sShop"),
-                "French": re.compile(r"Magasins\s(Kecleon)"),
-                "German": re.compile(r"(Kecleon)-Laden"),
-                "Italian": re.compile(r"Magazzini\s(?:\[CS:.])?(Kecleon)"),
-                "Spanish": re.compile(r"Tienda\s(Kecleon)"),
-                "Japanese": re.compile(r"(カクレオン)(?:\[CR])?の\s?お?みせ"),
-            }
-            kecleon_replace_regions = [
                 self.static_data.string_index_data.string_blocks.get(
-                    "Floor-Wide Status Names+Desc"
+                    "IQ Skills Descriptions"
                 ),
+            ]
+            # Some pokemons (Kecleon, Shaymin) need extra care, because some item long descriptions contain references to the shop (should be replaced) and the Pokemon itself (should not be replaced) at the same time.
+            # Instead we replace some of the mentions of the specific shop names (note: things like music track names should not be replaced).
+            shop_texts = {
```

Comment:
Previously this part was only used for Kecleon's Shop, however upon analysis it seems that Shaymin's Delivery Service is another phrase that should be replaced with care.